backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -264,40 +280,61 @@ func initDBClient(initConnectionTimeout time.Duration) *storage.DB {
 		glog.Fatalf("Failed to drop unique key on pipeline name. Error: %s", response.Error)
 	}
 
-	response = db.Model(&model.ResourceReference{}).ModifyColumn("Payload", "longtext not null")
+	response = db.Model(&model.ResourceReference{}).ModifyColumn("Payload", textFormat)
 	if response.Error != nil {
 		glog.Fatalf("Failed to update the resource reference payload type. Error: %s", response.Error)
 	}
 
 	response = db.Model(&model.Run{}).AddIndex("experimentuuid_createatinsec", "ExperimentUUID", "CreatedAtInSec")
-	if response.Error != nil {
+	if ignoreAlreadyExistError(response.Error) != nil {
 		glog.Fatalf("Failed to create index experimentuuid_createatinsec on run_details. Error: %s", response.Error)
 	}
 
 	response = db.Model(&model.Run{}).AddIndex("experimentuuid_conditions_finishedatinsec", "ExperimentUUID", "Conditions", "FinishedAtInSec")
-	if response.Error != nil {
+	if ignoreAlreadyExistError(response.Error) != nil {
 		glog.Fatalf("Failed to create index experimentuuid_conditions_finishedatinsec on run_details. Error: %s", response.Error)
 	}
 
 	response = db.Model(&model.Pipeline{}).AddUniqueIndex("name_namespace_index", "Name", "Namespace")
-	if response.Error != nil {
+	if ignoreAlreadyExistError(response.Error) != nil {
 		glog.Fatalf("Failed to create index name_namespace_index on run_details. Error: %s", response.Error)
 	}
 
-	response = db.Model(&model.RunMetric{}).
-		AddForeignKey("RunUUID", "run_details(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* update */)
-	if response.Error != nil {
-		glog.Fatalf("Failed to create a foreign key for RunID in run_metrics table. Error: %s", response.Error)
-	}
-	response = db.Model(&model.PipelineVersion{}).
-		AddForeignKey("PipelineId", "pipelines(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* update */)
-	if response.Error != nil {
-		glog.Fatalf("Failed to create a foreign key for PipelineId in pipeline_versions table. Error: %s", response.Error)
-	}
-	response = db.Model(&model.Task{}).
-		AddForeignKey("RunUUID", "run_details(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* update */)
-	if response.Error != nil {
-		glog.Fatalf("Failed to create a foreign key for RunUUID in task table. Error: %s", response.Error)
+	switch driverName {
```

Comment:
I think extracting the different behavior of each driver to their own module is helpful. But feel free to defer it to future PRs.

Reply:
I agree. This part was more of a patch work to make it work for now.